File: liqp-core/src/main/java/liqp/filters/Remove_First.java
package liqp.filters;

import java.util.regex.Pattern;
import liqp.nodes.RenderContext;

class Remove_First extends Filter {

    /*
     * remove_first(input, string)
     *
     * remove the first occurrences of a substring
     */
    @Override
    public Object apply(RenderContext context, Object value, Object... params) {

        String original = super.asString(value);

        Object needle = super.get(0, params);

        if (needle == null) {
            throw new RuntimeException("invalid pattern: " + needle);
        }

        return original.replaceFirst(Pattern.quote(String.valueOf(needle)), "");
    }
}

File: liqp-core/src/main/kotlin/liqp/filter/RemoveFirst.kt
package liqp.filter

import liqp.context.LContext
import java.util.regex.Pattern

class RemoveFirst : LFilter() {

  /**
   * remove_first(input, string)
   *
   * remove the first occurrences of a substring
   */
  override fun onFilterAction(context: LContext, value: Any?, params: FilterParams): Any? {
    val original = context.asString(value)
    val needle:Any = params[0] ?: throw RuntimeException("invalid pattern: " + params[0])

    return original?.replaceFirst(Pattern.quote(needle.toString()).toRegex(), "")
  }
}
